File: src/main/java/catwalks/util/meta/MetaValueProperty.java
package catwalks.util.meta;

/**
 * Created by TheCodeWarrior
 */
public abstract class MetaValueProperty<T> extends MetaProperty {
	
	public abstract void set(MetaStorage storage, T value);
	public abstract T get(MetaStorage storage);
	
	@Override
	public String getValue(MetaStorage storage) {
		return get(storage).toString();
	}
}

File: src/main/java/catwalks/util/meta/MetaValueProperty.kt
package catwalks.util.meta

/**
 * Created by TheCodeWarrior
 */
abstract class MetaValueProperty<T> : MetaProperty() {

    abstract operator fun set(storage: MetaStorage, value: T)
    abstract operator fun get(storage: MetaStorage): T

    override fun getValue(storage: MetaStorage): String {
        return get(storage).toString()
    }
}
